File: frontend/src/app/shared/service/admin/inventaire/InventaireAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {InventaireDto} from 'src/app/shared/model/inventaire/Inventaire.model';
import {InventaireCriteria} from 'src/app/shared/criteria/inventaire/InventaireCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class InventaireAdminService extends AbstractService<InventaireDto, InventaireCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): InventaireDto {
        return new InventaireDto();
    }

    public constrcutCriteria(): InventaireCriteria {
        return new InventaireCriteria();
    }

    get API() {
        return environment.apiUrlStockyservice + 'admin/inventaire/';
    }
}

File: backend-ms1/src/main/java/ma/zs/stocky/service/facade/admin/inventaire/InventaireAdminService.java
package ma.zs.stocky.service.facade.admin.inventaire;

import java.util.List;
import ma.zs.stocky.bean.core.inventaire.Inventaire;
import ma.zs.stocky.dao.criteria.core.inventaire.InventaireCriteria;
import ma.zs.stocky.zynerator.service.IService;



public interface InventaireAdminService extends  IService<Inventaire,InventaireCriteria>  {

    List<Inventaire> findByResponsableInvId(Long id);
    int deleteByResponsableInvId(Long id);
    long countByResponsableInvId(Long id);




}
